src/foam/core/pool/ThreadPoolAgency.js

```diff
@@ -283,6 +260,51 @@ foam.CLASS({
       Loggers.logger(x, this).info(getPrefix(), "available", getNumberOfThreads(), "queued", getQueued(), "waiting", getWaiting(), "executing", getExecuting(), "executed", getExecuted());
       scheduleReporting();
       `
+    },
+    {
+      name: 'initThreadPool',
+      type: 'Void',
+      visibility: 'protected',
+      javaCode: `
+        pool_ = new ThreadPoolExecutor(
+          getNumberOfThreads(),
+          getNumberOfThreads(),
+          10,
+          TimeUnit.SECONDS,
+          new LinkedBlockingQueue<Runnable>(),
+          new ThreadFactory() {
+            final AtomicInteger threadNumber = new AtomicInteger(1);
+
+            public Thread newThread(Runnable runnable) {
+              Thread thread = new Thread(
+                threadGroup_,
+                runnable,
+                getPrefix() + "-" + threadNumber.getAndIncrement(),
+                0
+                );
+              // Thread does not block server from shut down.
+              thread.setDaemon(true);
+              thread.setPriority(Thread.NORM_PRIORITY);
+              return thread;
+            }
+          }
+        );
+        pool_.allowCoreThreadTimeOut(true);
+
+      `
+    },
+    {
+      name: 'restart',
```

Comment:
The model implements COREService which provides start, stop, reload. So this method should be called reload. 
and you can move the shutdown/close to 'stop', then you have finer control. 
reload would call stop, initThreadPool.